lightning/src/util/persist.rs

```diff
@@ -0,0 +1,75 @@
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! Expose a simple key-value store trait KVStorePersister that allows one to 
+//! implement the persistence for ChannelManager, NetworkGraph, and 
+//! ChannelMonitor all in one place.
+
+use core::ops::Deref;
+use bitcoin::hashes::hex::ToHex;
+use prelude::*;
+use io::{self};
+
+use crate::{chain::{keysinterface::{Sign, KeysInterface}, self, transaction::{OutPoint}, chaininterface::{BroadcasterInterface, FeeEstimator}, chainmonitor::{Persist, MonitorUpdateId}, channelmonitor::{ChannelMonitor, ChannelMonitorUpdate}}, ln::channelmanager::ChannelManager, routing::network_graph::NetworkGraph};
+use super::{logger::Logger, ser::Writeable};
+
+/// Trait for a key-value store for persisting some writeable object at some key
```

Comment:
Given we rely on the specific paths downstream, both in `FilesystemPersister::read_channelmanager` and even downstream in the sample node (:sob:), we should document the paths here. Ideally we'd add a `read` method on `KVStorePersister` and then implement the loads here against that, but I don't think we need to do that in this specific PR given its not a new issue.

Reply:
#1427

Reply:
yeah, that's why I had at first added the key getters.  

I actually already implemented it in Sensei because it seemed like you didn't want in ldk.  I added a `read` and a `list` method since read_channelmonitors needs to be able to list all keys/filenames.

Here's what I did, let me know if you think it's roughly what you'd want to add to KVStorePersister for #1427 
https://github.com/L2-Technology/sensei/pull/37/files#diff-af422d8a9ab5e8778884033d55578f060f9d14dde012232f4faeb2b8a49cec92R12

Reply:
Right, I'd prefer we keep the keys as part of the `util::persist` module and abstract it away from the user instead of exposing them. That would imply, like you say, adding a `read` and `list` method in a followup and having utils to read the data. That is also important so that we can change the persistence behavior in the future, eg for #1426.